cli/src/stake.rs

```diff
@@ -1980,40 +1980,38 @@ pub fn process_split_stake(
     };
 
     let rent_exempt_reserve = if !sign_only {
-        if let Ok(stake_account) = rpc_client.get_account(&split_stake_account_address) {
-            if stake_account.owner == stake::program::id() {
-                return Err(CliError::BadParameter(format!(
+        let check_stake_account = |account: Account| -> Result<u64, CliError> {
+            match account.owner {
+                owner if owner == stake::program::id() => Err(CliError::BadParameter(format!(
                     "Stake account {split_stake_account_address} already exists"
-                ))
-                .into());
-            } else if stake_account.owner == system_program::id() {
-                if !stake_account.data.is_empty() {
-                    return Err(CliError::BadParameter(format!(
-                        "Account {split_stake_account_address} has data and cannot be used to split stake"
-                    ))
-                    .into());
+                ))),
+                owner if owner == system_program::id() => {
+                    if !account.data.is_empty() {
+                        Err(CliError::BadParameter(format!(
+                            "Account {split_stake_account_address} has data and cannot be used to split stake"
```

Comment:
i wonder if we should spruce up these error messages to be a little more normie-friendly? maybe in follow up

Reply:
Sure, I'll make a follow-up. If you have recommended verbiage, please do share.